### PR TITLE
gcp: add support for labeling resources

### PR DESCRIPTION
**Pull request by ixdy**
GCP supports labeling resources [1] to help group them together in a
lightweight way. These labels should not be confused with tags, which
are a different but related concept.

This change adds support for user-specified labels that should be
applied to GCP resources. Similar functionality in the installer is
already available for other cloud providers (such as AWS).

[1] https://cloud.google.com/compute/docs/labeling-resources

### CODE REVIEW

**Comment by abhinavdahiya** [comment]
Labels do not have enough widespread availability in terms of the resources suported for us to support these a platform level field.

And as this change is targeting the machines only. I would recommend that we support labels for our control plane and worker machine using https://github.com/openshift/installer/blob/master/docs/user/overview.md#multiple-invocations

So the workflow would be
```
openshift-install create manifests
# update the Machine and Machineset objects to have label fields
openshift-install create cluster
```

To achieve this for compute is simple as the machine-api will adds these labels when they create these.
Since the installer creates the control-plane machines, we need the terraform code to add these label. For that
* the terraform supports labels input for all machines
* the tfvars function for gcp has access to the machine objects for the control plane, so we use the labels set in those to set the input created in above.

This way we can support day-1 machines with labels untill we can accept label in GCP for much wide resource set.

**Comment by ixdy** [comment]
Thanks for the feedback.

I added this to the GCP platform configuration to match the `userTags` options available for the [AWS provider](https://github.com/openshift/installer/blob/cfecc2c28f8f74a32db37a7cd9745b1f4e68de10/docs/user/aws/customization.md#cluster-scoped-properties).

I'd initially focused on just the machines, since that was the only reference I found for the AWS `userTags` field.

I've since piped through the GCP labels to the Terraform variable `gcp_extra_labels`, and have added labels to all GCP resources that seem to support them. (You're right that they aren't universally supported, but they are still supported by quite a few resource types.)

(Additionally, it wasn't clear to me how I could modify the labels variable in the Terraform configuration without either adding the labels to the install config or adding a new flag to the installer, the latter of which seems suboptimal.)

**Comment by ixdy** [comment]
Friendly ping - any comments on the updates I've provided?

**Comment by abhinavdahiya** [comment]
> based on https://cloud.google.com/compute/docs/labeling-resources

```
Apply labels to any of these resources:

    Virtual machine instances
    Forwarding rules (beta)
    Images
    Persistent disks
    Persistent disk snapshots
    Cloud Storage buckets
    Static external IP addresses (beta)
    VPN tunnels (beta)

```

so technically we can add labels here in this PR (GA) to VMs, Images (we do not create them anymore), disks (these are usually configured to be deleted with the machine we don't really create any without a corresponding machine), storage bucket and DNS zone. so 3 resource types are benefitting from this addition of labels while all the other resource types will not have this label.

Currently the number of resources that cannot be labelled (that we care about) >> the ones that can be, so I do not think it is consistent or correct to add an API that makes it seems like we can achieve labeling all the resources for the clusters.
That's why i suggested that we can focus on just the VMs by mulltiple invocations support in the installer see https://github.com/openshift/installer/blob/master/docs/user/overview.md#multiple-invocations

TLDR;
```
openshift-install create manifests
# modify the machine(set)s in the $INSTALL_DIR/openshift directory to add label fields
openshift-install create cluster
```
The installer will read these objects and add the labels to the control-plane machines as it created them today, and the machine-api in the cluster will take care of adding the label to the machines which it is running in the cluster for the compute ones.
